modules/pel/peltool/parse_user_data.py

```diff
@@ -40,25 +40,35 @@ def parse(self) -> str:
         else:
             value = self.parseCustom()
 
+        # Catch if value is None, otherwise python crashes
+        if value == None:
+            value = ("PROBLEM with UNRECOGNIZED value from parser creatorID={} compID={} subType={} version={}"
+                     .format(self.creatorID, hex(self.compID), self.subType, self.version))
+            return json.dumps(value)
         return value
 
     def parseCustom(self) -> str:
         import importlib
         name = (self.creatorID.lower() + "%04X" % self.compID).lower()
         try:
             cls = importlib.import_module("udparsers." + name + "." + name)
-            mv = memoryview(self.data)
-            return cls.parseUDToJson(self.subType, self.version, mv)
+            if self.data:
```

Comment:
this would be protecting against an empty UserData section.  Is that what you saw?

Reply:
Yes, on a few error flows.